client/controller.js

```diff
@@ -22,44 +18,15 @@ const debug = debugFactory( 'calypso:controller' );
  * `server/bundler/loader`. Sections are free to either ignore it, or use it
  * instead of directly calling `page` for linking routes and middlewares in
  * order to be also usable for server-side rendering (and isomorphic routing).
- * `clientRouter` then also renders React elements contained in `context.primary`.
+ * `clientRouter` then also renders the element tree contained in `context.layout`.
  */
 export function clientRouter( route, ...middlewares ) {
-	page( route, ...[ ...middlewares, renderElements ] );
-}
-
-export function renderElements( context ) {
-	renderPrimary( context );
-	renderSecondary( context );
-}
-
-function renderPrimary( context ) {
-	const { path, primary } = context;
-
-	// FIXME: temporary hack until we have a proper isomorphic, one tree
-	// routing solution (see https://github.com/Automattic/wp-calypso/pull/3302)
-	// Do NOT do this!
-	const sheetsDomElement = startsWith( path, '/themes' ) &&
-		document.getElementsByClassName( 'themes__sheet' )[0];
-
-	if ( primary && ! sheetsDomElement ) {
-		debug( 'Rendering primary', context, path, primary );
-		ReactDom.render(
-			primary,
-			document.getElementById( 'primary' )
-		);
-	}
+	page( route, ...[ ...middlewares, renderLayout ] );
 }
 
-function renderSecondary( context ) {
-	if ( context.secondary === null ) {
-		debug( 'Unmounting secondary' );
-		ReactDom.unmountComponentAtNode( document.getElementById( 'secondary' ) );
-	} else if ( context.secondary !== undefined ) {
-		debug( 'Rendering secondary' );
-		ReactDom.render(
-			context.secondary,
-			document.getElementById( 'secondary' )
-		);
-	}
+function renderLayout( context ) {
```

Comment:
Maybe we'll turn this into an actual middleware for explicit invocation in `my-sites/themes/index.js`, or even call it implicitly from `client/controller#renderLayout`. We could then have individual middleware gradually populating `context.primary`, `secondary`, and `tertiary` similar to what we have now, i.e. e.g.

```js
'/design/type/:tier': [ multiSite, makeNavigation, makeSiteSelection, makeLayout ]
```

where `makeNavigation` and `makeSiteSelection` put their element trees into `context.secondary`.